nemo_aligner/data/nlp/builders.py

```diff
@@ -265,7 +265,19 @@ def build_dataset(index, name):
 
 
 def build_sft_dataset(data_cfg, tokenizer, num_samples, answer_only_loss=True, is_chat=True, special_tokens=None):
-    dataset_cls = GPTSFTChatDataset if is_chat else GPTSFTDataset
+    packed_sequence = data_cfg.get("packed_sequence", False)
+    dataset_kwargs = {}
+
+    ## NOTE: sequence packing currently not supported with chat dataset
+    if is_chat:
+        dataset_cls = GPTSFTChatDataset
+    elif packed_sequence:
+        dataset_cls = GPTSFTPackedDataset
+        dataset_kwargs = {"return_cu_seqlen": data_cfg.get("packed_sequence_return_cu_seqlen", True)}
```

Comment:
Any documentation needed for this config arg?

Reply:
There's not much documentation on the NeMo side, but I went ahead and copied over the documentation that does exist.